features/FEATURE_BLE/ble/BLETypes.h

```diff
@@ -324,6 +324,25 @@ struct byte_array_t {
         return _value;
     }
 
+    /**
+     * Returns true if every byte is equal to zero
+     */
+    bool is_all_zeros() {
```

Comment:
Could we move this outside the class ? It should be easy to have a free function that works for both `ArrayView`, `byte_array_t` and other types as long as the object passed in overload the subscript operator and offer a `size()` member function.